### ISSUE TITLE
AssetEditor: Cannot replace the Asset on clicking the Asset's title

### ISSUE THREAD

**Issue by lorenzulrich** [neos/neos-ui#3057]
### Description

The Neos UI "suggests" that you could replace an Asset in the AssetEditor by clicking its title and searching another file.

### Steps to Reproduce

1. In an AssetEditor property, select an Asset and apply.
2. Click the Asset (not the X button to the right!): The Asset is "removed" and the field is turned into a search field.
3. Start typing a search term.

#### Expected behavior

Matching Assets would be listed and are clickable.

#### Actual behavior

No Asset but the currently used one can be found or selected.

### Demo

https://user-images.githubusercontent.com/1816023/156387922-e4adf609-c21e-460d-b029-35f0442c73a4.mp4

### Comment

As mentioned above (and can be seen in the video), if the Asset is removed first using the X button, replacing the Asset works.
So it would be perfectly fine to just do nothing on clicking the current Asset.

### Affected Versions

I reproduced this in 4.3 and 7.3
